carbon-design-system/carbon-addons-iot-react
# feat(list): add `hasFastSearch` and example async search story

Closes #3078 

**Summary**

- Adds `hasFastSearch` prop to search object on List to allow search to only happen on Enter or Blur. This gives us the ability to do async searches on a List
- Adds an example story with async loading and search.

**Change List (commits, features, bugs, etc)**

- adds `hasFastSearch` to List and ListHeader
- adds logic to call onChange via onBlur/Enter or onClear
- adds tests to confirm
- adds story to demo async loading/search

**Acceptance Test (how to verify the PR)**

- go to with async loading and search
- type 8
- confirm loading indicator appears
- after 1.5seconds you should have only items with `8` in them

- go to basic (single column) story
- turn on search
- type `dj`
- confirm search happens immediately
- click clear search button
- turn off check the hasFastSearch knob
- type `dj`
- confirm search does not happen until typing enter

**Regression Test (how to make sure this PR doesn't break old functionality)**

- check list search story and confirm default experience is the same as before



**Things to look for during review**
